lib2/CMakeLists.txt

```diff
@@ -26,16 +26,32 @@ if (APPLE)
 elseif (WIN32)
     #TODO:  Should find these dlls not assume them.
     if (NOT USE_NATIVE_LIBRARIES)
+      if (${ARCH64BIT} EQUAL 1)
+        find_file (BOOST_REGEX_BIN "boost_regex-vc100-mt-1_46.dll" "${EXTERNALS_DIRECTORY}/boost/windows-x86_64-vc100/lib" NO_DEFAULT_PATH)
+      else()
         find_file (BOOST_REGEX_BIN "boost_regex-vc90-mt-1_44.dll" "${EXTERNALS_DIRECTORY}/boost/windows-i386-vc90/lib" NO_DEFAULT_PATH)
+      endif()
     endif (NOT USE_NATIVE_LIBRARIES)
-    find_file (ICU_DT_BIN "icudt36.dll" "${EXTERNALS_DIRECTORY}/icu/bin" NO_DEFAULT_PATH)
-    find_file (ICU_IN_BIN "icuin36.dll" "${EXTERNALS_DIRECTORY}/icu/bin" NO_DEFAULT_PATH)
-    find_file (ICU_UC_BIN "icuuc36.dll" "${EXTERNALS_DIRECTORY}/icu/bin" NO_DEFAULT_PATH)
-    find_file (OPENSSL_LIB_BIN "libeay32.dll" "${EXTERNALS_DIRECTORY}/openssl/win32/lib" NO_DEFAULT_PATH)
-    find_file (OPENSSL_SSL_BIN "ssleay32.dll" "${EXTERNALS_DIRECTORY}/openssl/win32/lib" NO_DEFAULT_PATH)
-    find_file (XALAN_C_BIN "Xalan-C_1_10.dll" "${EXTERNALS_DIRECTORY}/xalan/xalan-c/bin" NO_DEFAULT_PATH)
-    find_file (XALAN_MESSAGES_BIN "XalanMessages_1_10.dll" "${EXTERNALS_DIRECTORY}/xalan/xalan-c/bin" NO_DEFAULT_PATH)
-    find_file (XERCES_C_BIN "xerces-c_2_7.dll" "${EXTERNALS_DIRECTORY}/xalan/xerces-c/bin" NO_DEFAULT_PATH)
+    if (${ARCH64BIT} EQUAL 1)
+      find_file (ICU_DT_BIN "icudt54.dll" "${EXTERNALS_DIRECTORY}/icu/bin64" NO_DEFAULT_PATH)
+      find_file (ICU_IN_BIN "icuin54.dll" "${EXTERNALS_DIRECTORY}/icu/bin64" NO_DEFAULT_PATH)
+      find_file (ICU_UC_BIN "icuuc54.dll" "${EXTERNALS_DIRECTORY}/icu/bin64" NO_DEFAULT_PATH)
+      find_file (OPENSSL_LIB_BIN "libeay32.dll" "${EXTERNALS_DIRECTORY}/openssl/win64/lib" NO_DEFAULT_PATH)
```

Comment:
This pull request try to provide a quick fix to allow building ODBC-Connector required libraries on Windows 64bit environment.
I try to avoid any change/update for building HPCC clienttools on Windows 32bit.

Here are some more details about the questions:

1.  Even before this pull request the platform/clienttools only can be compiled with Visual Studio 9 2008 (32bit) with particular version of external libraries due to hard-coded libraries directories and versions.  This pull request doesn't overcome this limitation but just add support for compiling the ODBC-Connector required libraries on Visual Studio 9 2008 64bit environment.
There are some questions posted on HPCC forum for asking how to compiling HPCC on Windows. We haven't document the libraries requirements as well as what Visual Studio versions we support.
   I can open a new JIRA ticket to cover the compiler and external libraries version issue and targeted for 6.0


2.  I am aware ARCH64BIT is for compiler architect. There is a "osdir" variable to distinguish the boost_regext lib is for 64bit or 32bit.

3. For 64bit I am trying to use the latest third party libraires if there are not available in the original external libraries prepared by Gordon.
   I try to avoid any change for Windows 32bit build environment (platform/clienttools) which leave the libraries versions different between 32bit and 64bit.